Wox.Core/Updater.cs

```diff
@@ -52,31 +55,47 @@ public async Task UpdateApp()
                 return;
             }
 
-            var fr = u.FutureReleaseEntry;
-            var cr = u.CurrentlyInstalledVersion;
-            Log.Info($"|Updater.UpdateApp|Future Release <{fr.Formatted()}>");
-            if (fr.Version > cr.Version)
+            var newReleaseVersion = Version.Parse(u.FutureReleaseEntry.Version.ToString());
+            var currentVersion = Version.Parse(Constant.Version);
+
+            Log.Info($"|Updater.UpdateApp|Future Release <{u.FutureReleaseEntry.Formatted()}>");
+
+            if (newReleaseVersion <= currentVersion)
+            {
+                if (!silentIfLatestVersion)
+                    MessageBox.Show("You already have the latest Wox version");
+                m.Dispose();
+                return;
+            }
+            
+            try
+            {
+                await m.DownloadReleases(u.ReleasesToApply);
+            }
+            catch (Exception e) when (e is HttpRequestException || e is WebException || e is SocketException)
             {
-                try
-                {
-                    await m.DownloadReleases(u.ReleasesToApply);
-                }
-                catch (Exception e) when (e is HttpRequestException || e is WebException || e is SocketException)
-                {
-                    Log.Exception($"|Updater.UpdateApp|Check your connection and proxy settings to github-cloud.s3.amazonaws.com.", e);
-                    m.Dispose();
-                    return;
-                }
-
-                await m.ApplyReleases(u);
+                Log.Exception($"|Updater.UpdateApp|Check your connection and proxy settings to github-cloud.s3.amazonaws.com.", e);
+                m.Dispose();
+                return;
+            }
+            
+            await m.ApplyReleases(u);
+            
+            // Not needed when in portable mode
+            if(!IsPortableMode)
                 await m.CreateUninstallerRegistryEntry();
-
-                var newVersionTips = this.NewVersinoTips(fr.Version.ToString());
-                
-                MessageBox.Show(newVersionTips);
-                Log.Info($"|Updater.UpdateApp|Update success:{newVersionTips}");
+            
+            if (IsPortableMode)
+            {
+                var targetDestination = m.RootAppDirectory + $"\\app-{newReleaseVersion.ToString()}\\{Constant.PortableFolderName}";
+                FilesFolders.Copy(Constant.PortableDataPath, targetDestination);
```

Comment:
add exception handling (maybe recovery?) + right message to user when fail

Reply:
sure thing

Reply:
what will happen if the filefolders throws an exception?

Reply:
Updated to inform the user if the operation failed.